langchain/chains/llm.py

```diff
@@ -115,20 +128,37 @@ async def aprep_prompts(
 
     def apply(self, input_list: List[Dict[str, Any]]) -> List[Dict[str, str]]:
         """Utilize the LLM generate method for speed gains."""
-        response = self.generate(input_list)
-        return self.create_outputs(response)
+        response, prompts = self.generate(input_list)
+        return self.create_outputs(response, prompts)
 
     async def aapply(self, input_list: List[Dict[str, Any]]) -> List[Dict[str, str]]:
         """Utilize the LLM generate method for speed gains."""
-        response = await self.agenerate(input_list)
-        return self.create_outputs(response)
-
-    def create_outputs(self, response: LLMResult) -> List[Dict[str, str]]:
+        response, prompts = await self.agenerate(input_list)
+        return self.create_outputs(response, prompts)
+
+    def _get_final_output(self, text: str, prompt_value: PromptValue) -> Any:
+        result: Any = text
+        for guardrail in self.guardrails:
+            if isinstance(guardrail, OutputGuardrail):
+                try:
+                    result = guardrail.output_parser.parse(result)
+                    error = None
+                except Exception as e:
```

Comment:
yeah we could add a special type of error that people could raise to just crash (i think thats what youre getting at)

what would use case be?

Reply:
Yeah, although with logic inverted -- allow folks to specify what exceptions can be excepted if they want to rather than have to throw a specific exception for halting execution.

In my mind though AssertionErrors are special and generally not be caught as they indicate there's a logical error in the code and execution should halt.

The use case is to make sure that we don't hide a logical error in the parsing code. So when an exception is raised we can distinguish between parser input being malformed vs. the parser itself having a bug.

I imagine with the agents, things could get interesting as they may try to compensate for the parse bug

Reply:
A suggested diff to improve error handling
https://github.com/hwchase17/langchain/compare/harrison/guardrails...nc/guardrails-error-handling?expand=1

- ValidationError is now a subclass of Exception
- Each OutputParser can now declare its own subclass of ValidationError that people can reference in a catch statement